exp/runtime/api/v1beta1/extension_webhook_test.go

```diff
@@ -0,0 +1,182 @@
+}
+
+func TestExtensionConfigValidate(t *testing.T) {
+	extensionWithURL := &Extension{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-extension",
+		},
+		Spec: ExtensionSpec{
+			ClientConfig: ExtensionClientConfig{
+				URL: pointer.String("https://extension-address.com"),
+			},
+		},
+	}
+
+	extensionWithService := &Extension{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-extension",
+		},
+		Spec: ExtensionSpec{
+			ClientConfig: ExtensionClientConfig{
+				Service: &ServiceReference{
+					Path:      pointer.StringPtr("/path/to/handler"),
+					Port:      pointer.Int32(1),
+					Name:      "foo",
+					Namespace: "bar",
+				}},
+		},
+	}
+
+	// Valid updated Extension
+	updatedExtension := extensionWithURL.DeepCopy()
+	updatedExtension.Spec.ClientConfig.URL = pointer.StringPtr("https://a-in-extension-address.com")
+
+	badURLExtension := extensionWithURL.DeepCopy()
+	badURLExtension.Spec.ClientConfig.URL = pointer.String("https//extension-address.com")
+
+	extensionWithoutURLOrService := extensionWithURL.DeepCopy()
+	extensionWithoutURLOrService.Spec.ClientConfig.URL = nil
+
+	extensionWithInvalidServicePath := extensionWithService.DeepCopy()
+	extensionWithInvalidServicePath.Spec.ClientConfig.Service.Path = pointer.StringPtr("https://example.com")
+
+	extensionWithInvalidServicePort := extensionWithService.DeepCopy()
+	extensionWithInvalidServicePort.Spec.ClientConfig.Service.Port = pointer.Int32(90000)
+
+	tests := []struct {
+		name        string
+		in          *Extension
+		old         *Extension
+		featureGate bool
+		expectErr   bool
+	}{
+		{
+			name:        "creation should fail if feature flag is disabled",
+			in:          extensionWithURL,
+			featureGate: false,
+			expectErr:   true,
+		},
+		{
+			name:        "update should fail if feature flag is disabled",
+			old:         extensionWithURL,
+			in:          updatedExtension,
+			featureGate: false,
+			expectErr:   true,
+		},
+		{
+			name:        "update should fail if URL is invalid",
+			old:         extensionWithURL,
+			in:          badURLExtension,
+			featureGate: true,
+			expectErr:   true,
+		},
+		{
+			name:        "update should fail if URL and Service are both nil",
+			old:         extensionWithURL,
+			in:          extensionWithoutURLOrService,
+			featureGate: true,
+			expectErr:   true,
+		},
+
+		{
+			name:        "update should fail if Service Path is invalid",
+			old:         extensionWithService,
+			in:          extensionWithInvalidServicePath,
+			featureGate: true,
+			expectErr:   true,
+		},
+
+		{
+			name:        "update should fail if Service Port is invalid",
+			old:         extensionWithService,
+			in:          extensionWithInvalidServicePort,
+			featureGate: true,
+			expectErr:   true,
+		},
+		{
+			name:        "update should pass if Service Path is valid",
+			old:         extensionWithService,
+			in:          extensionWithService,
+			featureGate: true,
+			expectErr:   false,
+		},
```

Comment:
Can we add one test each for validating the name and namespace too?

Reply:
Will do - these tests aren't suppose to be comprehensive in this PR though. We can add a bunch of additional tests once we're happy with the design and level of validation.